src/libcore/hash/sip.rs

```diff
@@ -193,6 +193,14 @@ impl Hasher for SipHasher {
     }
 
     #[inline]
+    fn delimit(&mut self, len: usize) {
+        // skip the first delimiter
+        if self.length > 0 {
+            self.write_usize(len);
```

Comment:
Good points.

Yes I think `(&[x, 1], None)` and `(&[x], Some(0))` would both hash to the same sequence of x, 1, 0 (in for example 64-bit integers), That's not good, and it allows arbitrary many values to produce the same hash, so SipHasher cannot permit this.